pkg/generate/app/cmd/resolve.go

```diff
@@ -502,6 +503,11 @@ func AddMissingComponentsToRefBuilder(
 				errs = append(errs, fmt.Errorf("the Dockerfile in the repository %q has no FROM instruction", info.Path))
 				continue
 			}
+			err := exposedPortsAreNumeric(dockerfileutil.LastExposedPorts(node))
+			if err != nil {
+				errs = append(errs, fmt.Errorf("the Dockerfile has an invalid EXPOSE instruction: %v", err))
+				continue
+			}
```

Comment:
>is this also going to cause "oc new-build" to fail w/ these dockerfiles?

Yes, it will :-(

(`oc new-app` and `oc new-build` looks **very** similar. Do they duplicate each other?)